### PR TITLE
feat: add new theorem prod_div_rpow_distrib

### PR DESCRIPTION
**Pull request by casavaca**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/SpecialFunctions/Pow/Real.lean [hunk 1]
```diff
@@ -513,6 +513,19 @@ lemma rpow_mul_intCast (hx : 0 ≤ x) (y : ℝ) (n : ℤ) : x ^ (y * n) = (x ^ y
...
+    ∏ x in s, (f x / g x) ^ (h x) = (∏ x in s, (f x) ^ (h x)) / (∏ x in s, (g x) ^ (h x)) := by
+  revert hf hg
+  apply Finset.cons_induction_on (p := fun s => (∀ x ∈ s, 0 ≤ f x) → (∀ x ∈ s, 0 ≤ g x) →
+    ∏ x in s, (f x / g x) ^ (h x) = (∏ x in s, (f x) ^ (h x)) / (∏ x in s, (g x) ^ (h x)))
+  · simp only [implies_true, Finset.prod_empty, ne_eq, one_ne_zero, not_false_eq_true, div_self]
+  · simp only [Finset.prod_cons, Finset.mem_cons]
+    intro x s _ ih hf hg
+    rw [ih, div_rpow, div_mul_div_comm] <;>
+    intros <;> (first | apply hf | apply hg) <;>
+    first | (left; trivial) | (right; trivial)
```
Comment (j-loreaux) [inline Mathlib/Analysis/SpecialFunctions/Pow/Real.lean:527]: The proof simplifies to this, as it's basically just the combination of `Finset.prod_div_distrib` and `Real.div_rpow`.
```suggestion
  rw [← Finset.prod_div_distrib]
  congr! 1
  rename_i x hx
  rw [div_rpow (hf x hx) (hg x hx)] 
```

**Review by j-loreaux — COMMENTED** [review]
I'm sorry, [my comment](https://github.com/leanprover-community/mathlib4/pull/12120#discussion_r1565140267) on #12120 was too quick, and hence wrong, because I misread the statement.

I've left a simplified proof, as this really boils down to two existing lemmas, `Finset.prod_div_distrib` and `Real.div_rpow`. The former doesn't need any hypotheses in this case, and the latter only occurs *inside* the product.

Note that you would have the same issues with `Π x in s, (f x * g x) ^ (h x)`, because the fundamental difficulty is that the (real) power of the product of two real numbers is not the product of the powers, unless the two bases are nonnegative.

At it's heart (i.e., stripping away the lemmas that always apply), this result is just: `∏ x in s, (f x * g x) ^ (h x) = ∏ x in s, (f x) ^ (h x) * (g x) ^ (h x)`, but then we see that it has nothing to do with `∏` per se, because the same thing arises with `∑ x in s, (f x * g x) ^ (h x) = ∑ x in s, (f x) ^ (h x) * (g x) ^ (h x)`, or any other big operator.

So, for that reason, I don't think we actually want this lemma. Sorry for leading you astray on my comment to #12120. Does my explanation make sense?

File: Mathlib/Analysis/SpecialFunctions/Pow/Real.lean (hunk 1 above)
Comment (eric-wieser) [reply Mathlib/Analysis/SpecialFunctions/Pow/Real.lean:527]: I think you can use `congr! 1` with x hx` there

**Comment by casavaca** [comment]
Got it. Thank you @j-loreaux

**Pull request closed without merging**